core/includes/image.inc

```diff
@@ -526,6 +550,65 @@ function image_hex2rgba(&$hex) {
   );
 }
 
+/**
+ * Get an SVG image's defined dimensions.
+ *
+ * @param string $uri
+ *   The URI or path to an SVG image.
+ *
+ * @return array|FALSE
+ *   An array containing the keys "width" and "height" as integer values. If the
+ *   image is an SVG but no set dimensions are available, these keys will have
+ *   NULL values. If the image is not an SVG, or the SVG file is empty or
+ *   invalid, FALSE will be returned.
+ */
+function image_get_svg_dimensions($uri) {
+  // Safety check.
+  if (!image_is_svg($uri)) {
+    return FALSE;
+  }
+
+  // Return cached dimensions if already retrieved once this request.
+  $cached_images = &backdrop_static(__FUNCTION__, array());
```

Comment:
Out of interest: why static caching?

If it's called multiple times per request, it's probably called unnecessarily. (For example in the preprocessor.)

Reply:
Still seems to be called a couple times even though `image_get_svg_dimensions()` now only gets called once. So I think `image_get_info()` is called multiple times. Or I saw it with multiple images. I can't recall.

Reply:
Can get closed as resolved, then. :wink: